crates/anvil/src/eth/backend/db.rs

```diff
@@ -348,7 +348,7 @@ pub struct SerializableAccountRecord {
     pub nonce: u64,
     pub balance: U256,
     pub code: Bytes,
-    pub storage: BTreeMap<U256, U256>,
+    pub storage: BTreeMap<U256, B256>,
```

Comment:
this should be flipped, the key should always be b256

Reply:
I change also the key type

Reply:
That's good, but the value should stay U256

Reply:
If I the value is U256 the bug is not corrected:
https://github.com/foundry-rs/foundry/issues/8179
```
{
"0x0000000000000000000000000000000000000000000000000000000000000000":"0x1",
"0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc":"0x68e44eb31e111028c41598e4535be7468674d0a", // <--- The truncated zero
"0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103":"0x9e2803ad9ecc0a848c8a045329524d295f6ee44f",
"0xd30e835d3f35624761057ff5b27d558f97bd5be034621e62240e5c0b784abe68":"0x9965507d1a55bcc2695c58ba16fb37d819b0a4dc"}},
```